docker/Dockerfile

```diff
@@ -0,0 +1,128 @@
+ARG pkg_mgr=apt
+ARG make_cores=2
+
+FROM ubuntu:22.04 as common-base
+
+ENV TZ=America/Chicago
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+
+RUN apt update --fix-missing
+
+FROM common-base as apt-deps
+
+RUN apt install -y \
+        libssh-dev \
+        g++ \
+        gcc \
+        cmake \
+        make \
+        libxml2-dev \
+        libxml++2.6-dev \
+        libblas-dev \
+        liblapack-dev \
+        pkg-config \
+        coinor-libcbc-dev \
+        coinor-libclp-dev \
+        coinor-libcoinutils-dev \
+        coinor-libosi-dev \
+        libboost-all-dev \
+        libhdf5-dev \
+        libsqlite3-dev \
+        libpcre2-dev \
+        gettext-base \
+        xz-utils \
+        python3-setuptools \
+        python3-pytest \ 
+        python3-tables \
+        python3-pandas \
+        python3-jinja2 \
+        cython3 \
+        libwebsockets-dev \
+        python3-pprintpp \
+    && apt clean -y all
+
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+
+FROM common-base as conda-deps
+
+RUN apt install -y \
+        wget \
+        bzip2 \
+        ca-certificates \
+    && apt clean -y all
+
+RUN echo 'export PATH=/opt/conda/bin:$PATH' > /etc/profile.d/conda.sh && \
+    wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh && \
+    /bin/bash ~/miniconda.sh -b -p /opt/conda && \
+    rm ~/miniconda.sh
+
+ENV PATH /opt/conda/bin:$PATH
+
+RUN conda config --add channels conda-forge
+RUN conda update -n base -c defaults conda
+RUN conda install -y mamba
+RUN conda update -y --all && \
```

Comment:
So everything is being installed via apt and conda? why?

Reply:
Using the (standard) `buildx` environment, it only builds the stages that are necessary, based on a dependency check. The `cyclus` stage is based on a stage named `${pkg_mgr}-deps`.  In the case when `pkg_mgr == apt` it will only build the `apt-deps` stage and when `pkg_mgr == conda`` it will only build the `conda-deps` stag.

Reply:
That makes sense. That just wasn't clear to me.

Reply:
Out of curiosity, why are the apt and conda stages installing different versions of cython?

Reply:
That's a good question. Does this build if you have cython 3 installed? Previously I haven't been able to get it to build with that version.

Reply:
Alas, they are the same version.  The apt build uses the (poorly named) `cython3` package which is cython version v0.29 for python 3.  The conda version is the same.